### ISSUE TITLE
ITRF2000 file is strange

### ISSUE THREAD

**Issue by ValZapod** [OSGeo/PROJ#3250]
https://github.com/OSGeo/PROJ/blob/a390c57ab5998154535d7ffbc7de04aad2a73910/data/ITRF2000#L4-L6

No, it is not, it was false even back then. Also, +dx=0.0002 should be +dx=0.0003 (see link to itrf.ign.fr below).

epoch should be 2000.0 only, like in ITRF2014 or ITRF2008.

https://github.com/OSGeo/PROJ/blob/a390c57ab5998154535d7ffbc7de04aad2a73910/data/ITRF2000#L8-L12

These should not be same.

All of the paramaters if you cannot calculate those yourself, which is understandable, are there, forward and backward: https://itrf.ign.fr/en/solutions/transformations

**Comment by kbevers** [comment]
Please submit a pull request with a fix to this

**Comment by ValZapod** [comment]
Can I add <ITRF2014> too in this file? Also can I add ITRF2020 everywhere (that would require new file) or you want to add seismic (nee in ITRF2020) and tectonics first?

**Comment by rouault** [comment]
> Also, +dx=0.0002 should be +dx=0.0003 (see link to itrf.ign.fr below).

Looking with te wayback machine, the parameters we have currently are consistent with the ones of https://web.archive.org/web/20220308203942/http://itrf.ign.fr/ITRF_solutions/2005/tp_05-00.php
So it seems they have been updated since then.
However our current parameters are consistent with what is in EPSG:
```
$ bin/projinfo -s ITRF2000 -t ITRF2005 -o PROJ -q
+proj=pipeline
  +step +proj=axisswap +order=2,1
  +step +proj=unitconvert +xy_in=deg +xy_out=rad
  +step +proj=cart +ellps=GRS80
  +step +proj=helmert +x=-0.0001 +y=0.0008 +z=0.0058 +rx=0 +ry=0 +rz=0 +s=-0.0004
        +dx=0.0002 +dy=-0.0001 +dz=0.0018 +drx=0 +dry=0 +drz=0 +ds=-8e-05
        +t_epoch=2000 +convention=position_vector
  +step +inv +proj=cart +ellps=GRS80
  +step +proj=unitconvert +xy_in=rad +xy_out=deg
  +step +proj=axisswap +order=2,1
```
So the best action here would be to make sure they are updated in EPSG first, if that's the correct action. I'll contact IOGP about that.

> These should not be same.

The content of the ITRF2000 file is also consistent with what EPSG gives (signs are reversed due to the +inv, but end result is the same):
```
$ bin/projinfo -s ITRF2000 -t ITRF97 -o PROJ -q
+proj=pipeline
  +step +proj=axisswap +order=2,1
  +step +proj=unitconvert +xy_in=deg +xy_out=rad
  +step +proj=cart +ellps=GRS80
  +step +inv +proj=helmert +x=-0.0067 +y=-0.0061 +z=0.0185 +rx=0 +ry=0 +rz=0
        +s=-0.00155 +dx=0 +dy=0.0006 +dz=0.0014 +drx=0 +dry=0 +drz=-2e-05
        +ds=-1e-05 +t_epoch=1997 +convention=position_vector
  +step +inv +proj=cart +ellps=GRS80
  +step +proj=unitconvert +xy_in=rad +xy_out=deg
  +step +proj=axisswap +order=2,1

$ bin/projinfo -s ITRF2000 -t ITRF96 -o PROJ -q
+proj=pipeline
  +step +proj=axisswap +order=2,1
  +step +proj=unitconvert +xy_in=deg +xy_out=rad
  +step +proj=cart +ellps=GRS80
  +step +inv +proj=helmert +x=-0.0067 +y=-0.0061 +z=0.0185 +rx=0 +ry=0 +rz=0
        +s=-0.00155 +dx=0 +dy=0.0006 +dz=0.0014 +drx=0 +dry=0 +drz=-2e-05
        +ds=-1e-05 +t_epoch=1997 +convention=position_vector
  +step +inv +proj=cart +ellps=GRS80
  +step +proj=unitconvert +xy_in=rad +xy_out=deg
  +step +proj=axisswap +order=2,1

$ bin/projinfo -s ITRF2000 -t ITRF94 -o PROJ -q
+proj=pipeline
  +step +proj=axisswap +order=2,1
  +step +proj=unitconvert +xy_in=deg +xy_out=rad
  +step +proj=cart +ellps=GRS80
  +step +inv +proj=helmert +x=-0.0067 +y=-0.0061 +z=0.0185 +rx=0 +ry=0 +rz=0
        +s=-0.00155 +dx=0 +dy=0.0006 +dz=0.0014 +drx=0 +dry=0 +drz=-2e-05
        +ds=-1e-05 +t_epoch=1997 +convention=position_vector
  +step +inv +proj=cart +ellps=GRS80
  +step +proj=unitconvert +xy_in=rad +xy_out=deg
  +step +proj=axisswap +order=2,1

```

**Comment by rouault** [comment]
I've emailed feedback@iogp.org about that issue.

> Also can I add ITRF2020 everywhere (that would require a new file) or you want to add seismic (new in ITRF2020) and tectonics first?

I'm wondering if we should perpetuate those ITRFxxxx files (or if they are useful in such form, maybe automate their generation from PROJ database. This might be tricky if several transformations were filed, for example the original ones used, and potentially the updated ones with latest IGN data). They are predating PROJ database which contains ITRF transformations registered with EPSG. The ITRF2020 transformations are in the latest releases of proj.db
cf 
```
$ bin/projinfo -s ITRF2014 -t ITRF2020 -o PROJ -q
+proj=pipeline
  +step +proj=axisswap +order=2,1
  +step +proj=unitconvert +xy_in=deg +xy_out=rad
  +step +proj=cart +ellps=GRS80
  +step +proj=helmert +x=0.0014 +y=0.0009 +z=-0.0014 +rx=0 +ry=0 +rz=0 +s=0.00042
        +dx=0 +dy=0.0001 +dz=-0.0002 +drx=0 +dry=0 +drz=0 +ds=0 +t_epoch=2015
        +convention=position_vector
  +step +inv +proj=cart +ellps=GRS80
  +step +proj=unitconvert +xy_in=rad +xy_out=deg
  +step +proj=axisswap +order=2,1
```

**Comment by ValZapod** [comment]
>Looking with the wayback machine, the parameters we have currently are consistent with the ones of https://web.archive.org/web/20220308203942/http://itrf.ign.fr/ITRF_solutions/2005/tp_05-00.php

Where? +dx=0.0002 is not there... (That is the very left 2nd raw element, called T1.)

>The content of the ITRF2000 file is also consistent with what EPSG gives

Oh, yes, those are all the same. (Those are 2020 to 97, to 96, to 94.)

https://itrf.ign.fr/en/solutions/transformations?form%5BfromItrf%5D=ITRF2000&form%5BtoItrf%5D=ITRF97&form%5Bsubmit%5D=&form%5B_token%5D=072600d778e23d6b.T9t6tgDnYUKtB9hEXP4kiwBreNUgmn43HpT5SPrqzTo.PI82gHKWABjaRKkcMKlD73cFNaNu7E4Da_-_AMuto1walzWGZZ01E_gymQ

https://itrf.ign.fr/en/solutions/transformations?form%5BfromItrf%5D=ITRF2000&form%5BtoItrf%5D=ITRF96&form%5Bsubmit%5D=&form%5B_token%5D=2b071037eb2330f5a581af12fafe.W7PWam3rDCxyzmDDzWU5ATMLRXiObzqNgL9FRWml3LE.KOeaXB-abXYFjRGboTJeZURlCA7AGQq59dQDDVjistcO_5laCJFYfSf7IQ

https://itrf.ign.fr/en/solutions/transformations?form%5BfromItrf%5D=ITRF2000&form%5BtoItrf%5D=ITRF94&form%5Bsubmit%5D=&form%5B_token%5D=38498d74fc84c059b.MG_5fx-2N50Oo1w7iPJJMrGNYfATzVidz5gRsP_DQH4.Qzu1SW3HVsd54C1j5KUuVsbjLIZdu2ipuvNX-M6ELhhlI7ZPesxjzFuWHQ

>The ITRF2020 transformations are in the latest releases of proj.db

Does this use tectonics (Station position kinematic model)? I doubt it. Also there is post-seismic deformation (PSD), is that do be used by Proj?

**Comment by ValZapod** [comment]
> first value on the Rates line:

But all the other values are taken from first two rows...

>At least for that transformation, the values are the same as

Yep!

**Comment by rouault** [comment]
Preliminary answer from IOGP:
""""
Hello Even,

Thank you for your message. We have allocated this to change request
2022.055 – please include this reference in any further correspondence on
this issue.

I have verified that the parameter values for EPSG:6302 are consistent with
the original source information, 
 

I am not aware of any update to these parameter values so am unable to
explain why the ITRF web page gives a different value for the tX rate
parameter. We will need to contact IERS to understand. I have sent a message
to IGN.


For the other two transformations, the ITRF web page gives a different value
for the parameter reference epoch to the original information source.
However when converted to a common parameter reference epoch the changed
parameters equal those originally published and the two versions of the
transformation are equivalent.

We are aware of some of the oldest inter-ITRF transformations (e.g.
EPSG:9020) having published values that are not exactly consistent with
values that can be derived from later IERS publications, but in these cases
the differences are within the uncertainty estimates of the transformation
parameter values. Following IERS advice we retained the original values and
added remarks, but did not add a new transformation because there is no
official documentation giving the now-recommended derived values. These
transformations are likely of historic interest only.

At this time we have not studied any further transformations, but have no
reason to believe that any of the inter-ITRF transformations currently in
the EPSG Dataset are inconsistent with what IERS was publishing at the time
of the data being entered into the EPSG Dataset. We will await a response
from IGN/IERS.
"""

**Comment by RogerLott** [comment]
I have had confirmation from IERS that the official and authoritative information is at https://itrf.ign.fr/docs/solutions/itrf2020/Transfo-ITRF2020_TRFs.txt and that there has been a bug in the IERS calculation at https://itrf.ign.fr/en/solutions/transformations. IERS will be fixing this bug and make take the calculator down whilst this is being done.

As a consequence I can confirm that there is no error in any of the intra-ITRF transformations currently in the EPSG Dataset (v10.066) and no action will be needed with these.

However, the official information at https://itrf.ign.fr/docs/solutions/itrf2020/Transfo-ITRF2020_TRFs.txt can be used to derive equivalent transformations to those already in the EPSG Dataset. We have identified five transformations where the IERS values have been changed (EPSG codes 9020-23 and 9025). We will be creating new additional transformation entries superseding these. Work on this has commenced and they will be published as soon as they are through our QC procedures.

Roger Lott
for IOGP Geodesy Subcommittee
